compiler/treetab.nim

```diff
@@ -12,16 +12,18 @@
 import
   hashes, ast, astalgo, types
 
-proc hashTree(n: PNode): Hash =
-  if n == nil: return
+proc hashTree*(n: PNode): Hash =
+  if n.isNil:
+    return
   result = ord(n.kind)
   case n.kind
   of nkEmpty, nkNilLit, nkType:
     discard
   of nkIdent:
+    echo "what", "ident"
     result = result !& n.ident.h
   of nkSym:
-    result = result !& n.sym.name.h
+    result = result !& n.sym.id
```

Comment:
I recommended this change. 

What @alehander42 needed was a table being able to associate metadata to certain AST trees featuring symbols such as variables and fields. Treating symbols with the same name as equal seemed wrong and I expected that no code in the compiler is currently dependent on this behaviour.